x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -48,6 +49,11 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
 
         if (event.localNodeMaster()) {
+            if (XPackPlugin.isReadyForXPackCustomMetadata(event.state()) == false) {
+                logger.debug("cannot add ML metadata to cluster as the following nodes might not understand the ML metadata: {}",
+                    () -> XPackPlugin.nodesNotReadyForXPackCustomMetadata(event.state()));
+                return;
+            }
```

Comment:
Following the changes I made in #30751 there's no need to change this file in this PR.  We no longer eagerly install the ML metadata on startup.